### PR TITLE
Read new RAMSES info_rt format (compatible with old)

### PR DESCRIPTION
**Pull request by MartinAlvarezSergio**
New RAMSES-RT info files contain information about the RT precision. This can be read in a backwards-compatible manner with a simple try / except.

### CODE REVIEW

File: yt/frontends/ramses/field_handlers.py [hunk 1]
```diff
@@ -448,7 +448,12 @@ def read_rhs(cast):
             # Read nRTvar, nions, ngroups, iions
             for i in range(4):
                 read_rhs(int)
-            f.readline()
+            # Try to read rtprecision
+            try:
+		read_rhs(int)
+		f.readline()
+            except Exception:
+                pass
```
Comment (cphyc) [inline yt/frontends/ramses/field_handlers.py]: I am a bit concerned about these lines, as the `Exception` is likely to be raised after the line has been read (see line 443). As a consequence, if the code fails to parse the input as `rtprecision`, then you should rewind the line.

Also, why do you read an extra line after calling `read_rhs`? This is only required if the immediate line after `rtprecision` is blank.
I would suggest

```suggestion
            try:
                ipos = f.tell()
                read_rhs(int)
            except Exception:
                f.seek(ipos)
```

Comment (MartinAlvarezSergio) [reply yt/frontends/ramses/field_handlers.py]: Hi Corentin,

Thanks for looking at this. I wanted to make a small push before I start pushing other modifications I've done to my yT that might come useful to other people.

Certainly, I can share a fragment of one of the new files here. The top of a file with the new format looks as follows:
`nRTvar      =         12
nIons       =          3
nGroups     =          3
iIons       =         10
rtprecision =          8
 
X_fraction  =  0.759999990463257E+00
Y_fraction  =  0.239999994635582E+00
 
unit_np     =  0.462706068183299E+04
unit_pf     =  0.693930649464614E+13`

The only difference with the previous format is the addition of an extra line containing an integer for rtprecision. As you mention in the new format, the code will read the new line and then the following one, which is an empty line, so it does work in that instance. Alternatively, in the old version, the exception will be raised when trying to split the empty line, thus there is no need for a second readline. I thought this a simple way to make the update, as it works with both formats, and should be easy to update in the future. I have also tested that it correctly works with both type of files.

Comment (cphyc) [reply yt/frontends/ramses/field_handlers.py]: Oh I see, my bad! I hadn't seen you replaced `f.readline()` by your code snippet.

Comment (cphyc) [inline yt/frontends/ramses/field_handlers.py]: This should fix the tests (replacing tabs with spaces)

```suggestion
            # Try to read rtprecision
            try:
                read_rhs(int)
                f.readline()
            except Exception:
                pass
```

**Comment by Xarthisius** [comment]
@yt-fido test this please

**Comment by matthewturk** [comment]
So I think I talked myself into this being backward compatible, but it did require me thinking things through.  Is this safe, in that we'll always have a new int-parseable line in the format?  And, if so, can we catch the `ValueError` or whatever instead of `Exception` and put a comment in explaining that it'll only get called in the case that the line is blank?

**Review by cphyc — DISMISSED** [review]

**Comment by munkm** [comment]
I'd like to second @matthewturk 's question here.

File: yt/frontends/ramses/field_handlers.py [hunk 2]
```diff
@@ -448,7 +448,12 @@ def read_rhs(cast):
             # Read nRTvar, nions, ngroups, iions
             for i in range(4):
                 read_rhs(int)
-            f.readline()
+            # Try to read rtprecision
+            try:
+                read_rhs(int)
+                f.readline()
+            except Exception:
```
Comment (neutrinoceros) [inline yt/frontends/ramses/field_handlers.py:455]:
```suggestion
            except ValueError:
```

**Comment by cphyc** [comment]
> So I think I talked myself into this being backward compatible, but it did require me thinking things through. Is this safe, in that we'll always have a new int-parseable line in the format? And, if so, can we catch the `ValueError` or whatever instead of `Exception` and put a comment in explaining that it'll only get called in the case that the line is blank?

Either the variable `rtprecision` is set (see also https://github.com/yt-project/yt/pull/2505#discussion_r398180381), then the file looks like:
```
nGroups     =          3
iIons       =         10
rtprecision =          8
 
X_fraction  =  0.759999990463257E+00
```
or the variable does not exist and the file reads
```
nGroups     =          3
iIons       =         10
 
X_fraction  =  0.759999990463257E+00
```
In the first case, the line with `rtprecision` is read, split into two, and the r.h.s. is parsed as an int, the code then reads an extra empty line.
In the second case, the code reads the line directly after `ilons=...` (which is empy), and fails at splitting it into two parts raising a ValueError.

Does this answer the question?

**Comment by neutrinoceros** [comment]
I'll close + reopen to refresh tests

**Comment by matthewturk** [comment]
declare_approval

**Review by matthewturk — APPROVED** [review]

**Comment by matthewturk** [comment]
@MartinAlvarezSergio thank you for your contribution!

**Review by neutrinoceros — APPROVED** [review]

**Pull request merged**